src/net.rs

```diff
@@ -341,12 +367,13 @@ impl NetworkConnector for HttpConnector {
                 let stream = try!(SslStream::new(&context, stream));
                 Ok(HttpStream::Https(stream))
             },
-            _ => {
+            x => {
                 Err(io::Error::new(io::ErrorKind::InvalidInput,
-                                "Invalid scheme for Http"))
+                                format!("Invalid scheme for Http: {}", x)))
```

Comment:
Whatcha think of adding a `debug!` (or info?) here explaining that enabling the feature will let you make https requests?